### ISSUE TITLE
Problems about overlap in reference line

### ISSUE THREAD

**Issue by 1Jyfc** [ApolloAuto/apollo#14218]
### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 18.04)**: Linux Ubuntu 18.04
- **Apollo installed from (source or binary)**: source
- **Apollo version (3.5, 5.0, 5.5, 6.0)**: 6.0
- **Output of `apollo.sh config` if on `master` branch**:

### Steps to reproduce the issue:
Hello Apollo team, I've got some problems while reading codes of Apollo, and I failed to find something useful in codes by myself. So I wanna ask for some help, hope for your reply.

I am working on planning module recently, and I tried to trigger some abnormal stage transitions that transfer to LAND_FOLLOW but are not the last stages of corresponding scenarios.  And I find these transition are always triggered by incorrect overlap on reference line, when there is no certain type of overlap on reference line, or the two overlap ids mismatch, scenario will change into LAND_FOLLOW. But I cannot find where overlaps are calculated. I hope your help to find it.

Besides, in my understanding overlap represents a part of path with a certain property, so it is meaningful only when the ego is heading for it. So will it disappear when the ego pass it? (while the end_s of overlap is smaller than s of the ego)

Hope for help sincerely, thanks a lot!

**Comment by 1Jyfc** [comment]
Hope for your reply. It's important for me and several days passed I don't receive your reply, could you help me?
Besides, these days I've found that, in case of traffic light scenarios, traffic light overlaps are divided into two groups--done or not, and they seem to be decided by messages come from Perception module. Since I use SVL to simulate and its 3D ground truth replaces Perception module, I wonder how to decide a traffic overlap is done or not. Hope for reply, thanks very much.

**Comment by daohu527** [comment]
> so it is meaningful only when the ego is heading for it. So will it disappear when the ego pass it?

yes. It will look for the traffic sign ahead, you can ref to `GetFirstOverlap` to find out it.
https://github.com/ApolloAuto/apollo/blob/266afbf68d83fa6fac7a812ff8a950223f5ab2c0/modules/planning/common/reference_line_info.cc#L213-L218

**Comment by 1Jyfc** [comment]
OK here it is, thanks!
